cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -1014,13 +1014,24 @@ func (s *Server) fetchChartWithRegistrySecrets(ctx context.Context, chartDetails
 	chartID := fmt.Sprintf("%s/%s", appRepo.Name, chartDetails.ChartName)
 	log.Infof("fetching chart %q with user-agent %q", chartID, userAgentString)
 
+	// Look up the cachedChart cached in our DB to populate the tarball URL
+	cachedChart, err := s.manager.GetChartVersion(chartDetails.AppRepositoryResourceNamespace, chartID, chartDetails.Version)
+	if err != nil {
+		return nil, nil, status.Errorf(codes.Internal, "Unable to fetch the chart %s (version %s) from the namespace %q: %v", chartID, chartDetails.Version, chartDetails.AppRepositoryResourceNamespace, err)
+	}
+	var tarballURL string
+	if cachedChart.ChartVersions != nil && len(cachedChart.ChartVersions) > 0 && cachedChart.ChartVersions[0].URLs != nil {
```

Comment:
Given that we're requesting a particular version from the cache, why do we not check here that the `cachedChart.ChartVersions == 1` , rather than ` > 0`? Leaves me wondering whether we can get different values here?

Reply:
Yep, I wasn't totally sure the query returns always a single chartVersion (albeit it should), that's why relaxed the condition. But, yep, we better set it to `1` so that we can detect if more than one result is returned.